source/code/source/playfab/PlayFabWinHttpPlugin.cpp

```diff
@@ -372,10 +370,18 @@ namespace PlayFab
     void PlayFabWinHttpPlugin::SetPredefinedHeaders(const CallRequestContainer& requestContainer, HINTERNET hRequest)
     {
         UNREFERENCED_PARAMETER(requestContainer);
-        WinHttpAddRequestHeaders(hRequest, L"Accept: application/json", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"Content-Type: application/json; charset=utf-8", -1, 0);
-        WinHttpAddRequestHeaders(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str(), -1, 0);
-        WinHttpAddRequestHeaders(hRequest, L"X-ReportErrorAsSuccess: true", -1, 0);
+        if(!TryAddHeader(hRequest, L"Accept: application/json") ||
+            !TryAddHeader(hRequest, L"Content-Type: application/json; charset=utf-8") ||
+            !TryAddHeader(hRequest, (L"X-PlayFabSDK: " + std::wstring(PlayFabSettings::versionString.begin(), PlayFabSettings::versionString.end())).c_str()) ||
+            !TryAddHeader(hRequest, L"X-ReportErrorAsSuccess: true"))
+        {
+            setPredefinedHeadersResult = GetLastError();
```

Comment:
you are setting this variable with a win32 system error and not an hresult. You need to use HRESULT_FROM_WIN32 to convert it to an hresult

Reply:
That came in with the successive change to TryAddHeader (the HRESULT_FROM_WIN32 is called in there)

---
In reply to: [414066496](https://github.com/PlayFab/XPlatCoreTemplate/pull/93#discussion_r414066496) [](ancestors = 414066496)